### PR TITLE
Add gosec to the static tests

### PR DESCRIPTION
**Pull request by nixpanic**
# Describe what this PR does #

`gosec` is a security scanning tool that can be used similar to `go vet` for standard security concerns. This PR addresses the reported issues and adds `gosec` to the standard tests.

### CODE REVIEW

**Comment by nixpanic** [comment]
@humblec would you like to include this enhancement?

**Comment by humblec** [comment]
> @humblec would you like to include this enhancement?

@nixpanic Why not? lets try to get it in :) . Can you please check why the CI fails?

**Comment by nixpanic** [comment]
The failure in CI looks unrelated:
```Aug 30 11:15:57.555: INFO: Running '/usr/local/bin/kubectl --kubeconfig=/home/travis/.kube/config create -f https://raw.githubusercontent.com/rook/rook/master/cluster/examples/kubernetes/ceph/pool-test.yaml'
Aug 30 11:15:58.366: INFO: stderr: ""
Aug 30 11:15:58.366: INFO: stdout: "cephblockpool.ceph.rook.io/replicapool created\n"
Aug 30 11:15:58.375: INFO: ExecWithOptions {Command:[/bin/sh -c ceph fsid] Namespace:rook-ceph PodName:rook-ceph-tools-76c7d559b6-7cfds ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true}
Aug 30 11:15:58.375: INFO: >>> kubeConfig: /home/travis/.kube/config
Aug 30 11:15:59.351: INFO: stdErr occurred 
%!(EXTRA string=unable to get monitor info from DNS SRV with service name: ceph-mon
[errno 2] error connecting to the cluster
)[AfterEach] RBD
```

This is the same error that @Madhu-1 hit in #580.

File: scripts/gosec.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -o pipefail
+
+if [[ -x "$(command -v gosec)" ]]; then
+  find cmd pkg -type d -print0 | xargs --null gosec -severity=medium
```
Comment (Madhu-1) [inline scripts/gosec.sh]: can we test all severity levels?

Comment (nixpanic) [reply scripts/gosec.sh]: Sure we can, doing so in the next revision.

**Comment by Madhu-1** [comment]
now we have  one more issue in CI
```
Total download size: 72 M
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
http://download.ceph.com/rpm-nautilus/el7/x86_64/librados2-14.2.3-0.el7.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below wiki article 
https://wiki.centos.org/yum-errors
If above article doesn't help to resolve this issue please use https://bugs.centos.org/.
Error downloading packages:
  2:librados2-14.2.3-0.el7.x86_64: [Errno 256] No more mirrors to try.
The command '/bin/sh -c rm -f /etc/yum.repos.d/ceph-iscsi.repo && yum -y update && yum clean all' returned a non-zero code: 1
Makefile:58: recipe for target 'image-cephcsi' failed
make: *** [image-cephcsi] Error 1
```

File: pkg/rbd/nodeserver.go [hunk 2]
```diff
@@ -153,6 +153,7 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 	}
 	isMounted = true
 
+	// #nosec - allow anyone to write inside the target path
```
Comment (Madhu-1) [inline pkg/rbd/nodeserver.go]: staging target path

**Pull request merged**